synapse/handlers/device.py

```diff
@@ -452,6 +455,14 @@ async def check_device_registered(
 
         raise errors.StoreError(500, "Couldn't generate a device ID.")
 
+    async def _prune_too_many_devices(self, user_id: str) -> None:
+        """Delete any stale devices this user may have."""
```

Comment:
Echoing [Rich's comment](https://github.com/matrix-org/synapse/pull/14038#issuecomment-1274618835).
It's mildly confusing that we have two concepts of stale devices in the codebase. Let's avoid using the same term to refer to both.

```suggestion
        """Delete any excess old devices this user may have."""
```